addons/membership/models/membership.py

```diff
@@ -43,7 +42,9 @@ class MembershipLine(models.Model):
              "-Invoiced Member: A member whose invoice has been created.\n"
              "-Paid Member: A member who has paid the membership amount.")
 
-    @api.depends('account_invoice_line.invoice_id.state','partner.membership_state')
+    @api.depends('account_invoice_line.invoice_id.state',
+                 'account_invoice_line.invoice_id.payment_ids'
```

Comment:
IIRC that one is redundant, the dependency tracking puts triggers on all fields in the path, so `account_invoice_line.invoice_id.payment_ids.invoice_ids.type` will implicitly depend on `account_invoice_line.invoice_id.payment_ids`.